Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -15242,16 +15242,31 @@ void SpeculativeJIT::compileObjectAssign(Node* node)
     switch (node->child2().useKind()) {
     case ObjectUse: {
         SpeculateCellOperand source(this, node->child2());
+        GPRTemporary scratch1(this);
 
         GPRReg targetGPR = target.gpr();
         GPRReg sourceGPR = source.gpr();
+        GPRReg scratch1GPR = scratch1.gpr();
 
         speculateObject(node->child2(), sourceGPR);
 
         flushRegisters();
+
+        JumpList doneCases;
+        JumpList genericCases;
+
+        genericCases.append(branchIfNotType(sourceGPR, FinalObjectType));
+        genericCases.append(branchTest8(NonZero, Address(sourceGPR, JSObject::indexingTypeAndMiscOffset()), CCallHelpers::TrustedImm32(IndexingShapeMask)));
+        emitLoadStructure(vm(), sourceGPR, scratch1GPR);
+        constexpr unsigned structureFlags = Structure::s_hasAnyKindOfGetterSetterPropertiesBits | Structure::s_hasReadOnlyOrGetterSetterPropertiesExcludingProtoBits | Structure::s_dictionaryKindBits | Structure::s_hasUnderscoreProtoPropertyExcludingOriginalProtoBits;
```

Comment:
I wish we could somehow keep this flags in sync, e.g. `Structure::objectAssignAndDestructuringSlowPathFlags` or something, because I they are kinda subject to change.

Reply:
We removed this check since propertyHash check is enough.